app/models/theme_field.rb

```diff
@@ -568,6 +568,12 @@ def invalidate_baked!
     if (will_save_change_to_value? || will_save_change_to_upload_id?) && !will_save_change_to_value_baked?
       self.value_baked = nil
     end
+    if will_save_change_to_upload_id?
+      if upload && upload.extension == "js"
```

Comment:
`upload.extension` is not 100% reliable because it comes from the filename of the **first** upload with that SHA. e.g. if I uploaded a file with `.txt`, and then later uploaded the same file with `.js`, `upload.extension` would remain `.txt`.

Can we pull the extension from somewhere else? Maybe from the `about.json` reference?

Reply:
Not sure if we have it stored even? Plus you could be not using the CLI anyway and have the same problem by creating an upload. I think the risk is somewhat acceptable here, even though it is not ideal.